catroid/src/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -354,6 +354,10 @@ private Object interpretFunction(Functions function, Sprite sprite) {
 				return doubleValueOfLeftChild == null ? 0d : java.lang.Math.toDegrees(Math.atan(doubleValueOfLeftChild));
 			case EXP:
 				return doubleValueOfLeftChild == null ? 0d : java.lang.Math.exp(doubleValueOfLeftChild);
+			case POWER:
+				return ((doubleValueOfLeftChild == null || doubleValueOfRightChild == null)
+						|| (doubleValueOfLeftChild < 0 && doubleValueOfRightChild < 1 && doubleValueOfRightChild > -1))
```

Comment:
NaN is a valid return value of the `Formula.interpretFunction` function.

The compute button will show ERROR because children have no idea what NaN is.
In the stage, if a formula evaluates to NaN, the action is completely ignored. At least if the formula should be a number like in the SetXAction.

So, using following code snipped is fine
```java
case POWER:
    return (doubleValueOfLeftChild == null || doubleValueOfRightChild == null) ? 0d : java.lang.Math.pow(doubleValueOfLeftChild, doubleValueOfRightChild);
```

Btw, `Math.pow` doesn't throw any exception :wink: